command/agent/testagent.go

```diff
@@ -147,9 +148,13 @@ func (a *TestAgent) Start() *TestAgent {
 			panic(fmt.Sprintf("Error creating agent: %s", err))
 		}
 
-		agent.id = id
-		agent.LogOutput = a.LogOutput
+		logOutput := a.LogOutput
+		if logOutput == nil {
+			logOutput = os.Stderr
+		}
+		agent.LogOutput = logOutput
 		agent.LogWriter = a.LogWriter
+		agent.logger = log.New(logOutput, id, log.LstdFlags)
```

Comment:
Hmm they are there now that I'm looking. Perhaps just some of the tests are missing the prefix.